src/lib/cfa/hosts.rb

```diff
@@ -18,6 +29,11 @@ def initialize(file_handler: nil)
       super(AugeasParser.new("hosts.lns"), PATH, file_handler: file_handler)
     end
 
+    def initialize_clone(source)
+      # Cloning AugeasTree
+      @data = source.data.clone
```

Comment:
this won't work for more deeply nested trees. In general I recommend to not redefine it here. Also maybe it fits better in CFA itself. And last but not least in general there can be problem if both clones writes, as it shared parser cache. ( but  your usecase is to only compare read-only copy and modified one, so it should be fine ).

btw name clone is a bit confusing as ruby clone is also just one level. So I recommend to name it also deep_clone similar to deep_copy. See https://medium.com/@raycent/ruby-clone-vs-dup-8a49b295f29a

Reply:
that is why I've defined ```initialize_clone``` hook even for ```AugeasTree``` [here](https://github.com/yast/yast-network/pull/711/files/9f5243baf40b0e664ec9343bbed971349b071727#diff-2412af142fac00805b68b41fae60a699R10)

regarding renaming. I do not understand here. AFAIK ```initialize_clone``` is standard ruby hook so I can hardly choose its name. Isn't it so?

Reply:
ah, so it using hook. Then ok. I do not know about that ruby hook for cloning. So fine for me.

Reply:
thanks